src/main/java/wolox/training/models/User.java

```diff
@@ -0,0 +1,134 @@
+package wolox.training.models;
+
+import static wolox.training.utils.ErrorConstants.BOOK_ALREADY_OWNED;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import wolox.training.exceptions.BookAlreadyOwnedException;
+
+@Entity
+@Table(name = "t_user")
+public class User {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "USER_SEQ")
+    @SequenceGenerator(name = "USER_SEQ", sequenceName = "USER_SEQ")
+    private Long id;
+
+    @NotNull
+    @Column(nullable = false)
+    private String username;
+
+    @NotNull
+    @Column(nullable = false)
+    private String name;
+
+    @NotNull
+    @Column(nullable = false)
+    private LocalDate birthdate;
+
+    @NotNull
+    @Column(nullable = false)
+    @ManyToMany
```

Comment:
In the **@manytomany** tag we must indicate the type of cascade we want to use. Since by not indicating one, it will not do any function in this regard and it will bring you problems when interacting with the db.

It is usually recommended to use Casacade.ALL. But this is not always correct.
By saying **ALL**, you are saying: Persist, Merge, Refresh, Delete and Detach.

**Detach** is only so that something gets detached along the cascade when you detach an object from the persistence context manually. This almost never happens, so we won't focus on this point.

**Persist** says: If you persist the Users instance to the DB, then persist every item that this collection has in the DB as well.

**Delete** says: If you delete the Users instance from the DB, then delete every book item that this collection has in the DB as well.

**Merge** says: If you changed the in-memory values of the collection, then merge this changes to the DB so these are reflected there.

**Refresh** says: If something changed in the database values of the collection, then refresh those changes in the in-memory values.

Here is a [link](https://www.baeldung.com/jpa-cascade-types) explaining it.

In this case, we just need MERGE and REFRESH. [@ManyToMany(cascade = {CascadeType.REFRESH, CascadeType.MERGE})]
You can look at MERGE as the instruction to "merge" changes on the model back to the database.
But we're lacking the one that brings modifications iof the database to the model. And that's REFRESH :)